### ISSUE TITLE
Tractors spinning in circles

### ISSUE THREAD

**Issue by AutisticWiz** [Courseplay/courseplay#3651]
I am in a map with tight fields, but not so tight that it needs turn on field. When I manually set turn width they spin in circles until I manually correct it. I know this was a problem in 2017 as well. https://youtu.be/7Q-czxv8V9k

**Comment by pvaiko** [comment]
If it works with the auto turn radius then why on earth do you set it manually?

**Comment by Tensuko** [comment]
Couldn't you just turn off the 'turn on field' option ?

**Comment by cwattyeso** [comment]
I noticed an issue just now in game, when Harvesting a field, I have the 'Turn of Field' deactivated, yet the harvester would lift up and make it's turn right on the field edge, almost as if it was afraid to drive off the field at all. This meant it missed little patches of crops as it made this turn. I then tried changing the Turn Width thinking maybe that was the issue was that it was set too low, and instead what happened then was this:
![vlcsnap-2019-04-18-11h33m34s292](https://user-images.githubusercontent.com/13921716/56355326-eafab680-61cd-11e9-94be-dbcc28c95387.png)

It completely missed strips of the field then, as if me increasing the turn radius/width meant it couldn't get back on the course it was meant to follow, which if it drove off the field and made it's loop it could have done. Also when it reached the end of the first two rows it did the whole spin in circles either forward or reverse and had to have me manually stop it and put it back on course before letting the driver carry on.

Also for some very weird reason at the end of the course, when the Harvester became full, it didn't stop and request to be unloaded, but carried on driving the course right to the end, resulting in it missing the last lane and a half of crop.

**Comment by Tensuko** [comment]
@cwattyeso 
kinda different problem but what did you set up, like the whole course and stuff.... doubt this is a CP Issue....

**Comment by cwattyeso** [comment]
@Tensuko Field is a plain rectangle shape, no objects or anything anywhere near any of the sides. In the Combine Courseplay Settings, I went into Course Generator and left everything at automatic for Start Location/Direction and Width (9m with this header) set it to do just Up & Down Lanes no headlands and start from first waypoint.
Only thing I changed before starting the driver off was making sure Turn on Field was Deactivated which had no effect whatsoever when the worker did the course as he made his turns right on the field edge line each time. I also enabled the option to have the Combine stop whilst unloading. However again, when the tank got full with a lane & a half left to harvest, it just kept driving the course without stopping or asking to be unloaded, but obviously not harvesting.

The Combine Circle issue seemed to be a result of the course generation. For whatever reason the first and second lanes were practically right on top of each other, so after harvesting the first lane, the course wanted the harvester to turn round and come back down the lane it had just done, and I think this is where it got into it's stuck doing circles phase.

I did manage to capture it on a recording, so have it as a YouTube video, link to where I noticed the Harvester not harvesting correctly [https://youtu.be/q-CoVh5BUCQ?t=2819](url)

**Comment by Tensuko** [comment]
@cwattyeso a logfile would be nice too.
Guess Debugchannel 9 and 14 would be enough here.
Video not available, I look at it later.

**Comment by sootysax** [comment]
> If it works with the auto turn radius then why on earth do you set it manually?

because sometimes the auto turn radius makes wider turns than is actually needed and has to reverse on each row and takes way longer than it should than if you set the turn to the manual turn radius to the working width or less, and then the turns are 3x faster

**Comment by Tensuko** [comment]
And ends up in a bad worked field...

Also, for that faster turn, you got the 1 Headland + spiral or circle strategy option in the new Coursgenerator. There are no turns.

**Comment by cwattyeso** [comment]
I like the idea of doing 1 Headland + Spiral with Harvesters as it would be very handy to keep the pipe out of the crop for unloading purposes (until Mode 2 is implemented.) I did a course on a field, set the Harvester up to work Clockwise, start on headland, then move to up and down rows with spiral method. The Headland course it did perfect, pipe was on the outside of the crop the whole time, meant I was able to unload it easy when it went round the outside of the field. Then when it moved to the spiral up and down rows, it seemed to switch direction, as now on every single row the harvesters pipe was into the next row of crop, making it impossible to unload with crop destruction on as you'd need to drive through the crop. Not sure why that happens, the only way to correct it was to run the course counter-clockwise but then whilst it's doing the headland you can't unload it.

**Comment by Tensuko** [comment]
Yea that I forgot to mention. From headland to Mid Part, there is a direction change. You would have to do it other way round then you think. And if you got for Headlands, better don't start in a corner. You got to try it out a bit, as I found out, that some course settings in combination can't chang the direction, but the position where you start can do it.
We know about that, but not on our priority list. We will if can tweak that a bit.

**Comment by cwattyeso** [comment]
Yeah for Headlands I always make sure the starting point or ending point depending whether you're doing headlands before or after the up and down rows is in the middle of a nice relatively straight edge.

I quite like using the Spiral option on a lot of courses, though notice that on Rectangular Fields where you maybe wouldn't do a headland, the courseplay workers always develop that fear of driving off the field at the end of the row and then sweeping round to start the lane on the opposite side of the field. They instead still want to turn/reverse and drive all the time right down the field edge line. This then causes it to leave quite a few little patches of unworked areas where it picks up the tool makes this turn and then when reaching the next row making its turn and putting down the tool.

**Comment by Tensuko** [comment]
![fsScreen_2019_04_20_17_54_11](https://user-images.githubusercontent.com/27344744/56459546-6bb7df00-6395-11e9-996d-afcc1763cc83.png)
No need Spiral, no need adjust turn, just 2 Headland + Turn on Field off....

**Comment by Tensuko** [comment]
Closed due to inactivity, there is a lot changed since that version.
If faceing anything similar. Make a new Issue with debugchannel 12 and 14 on.
Closed.
